### PR TITLE
[main] Update dependencies from dotnet/runtime dotnet/arcade dnceng/internal/dotnet-optimization dotnet/xharness dotnet/hotreload-utils

### PR DESCRIPTION
**Pull request by ViktorHofer**
Another attempt of #54060

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -146,73 +146,73 @@
...
+      <Sha>af5c238556e204583b129cc8f5c7338f84dc2c40</Sha>
     </Dependency>
-    <Dependency Name="System.Runtime.CompilerServices.Unsafe" Version="6.0.0-preview.6.21307.1">
+    <Dependency Name="System.Runtime.CompilerServices.Unsafe" Version="6.0.0-preview.6.21314.1">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>1d2aa13ffc256a8c3f645de9c4d0f1d4f3296711</Sha>
+      <Sha>af5c238556e204583b129cc8f5c7338f84dc2c40</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21308.1">
+    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21310.3">
```
Comment (hoyosjs) [inline eng/Version.Details.xml:177]:
```suggestion
    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21308.1">
```

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -146,73 +146,73 @@
...
       <Uri>https://github.com/mono/linker</Uri>
-      <Sha>802aa459394baa2451ffacc02219ba54ef77ca44</Sha>
+      <Sha>caeaf2a3fb3f636805fdd4881df4f9a539fff8f6</Sha>
```
Comment (hoyosjs) [inline eng/Version.Details.xml:179]:
```suggestion
      <Sha>802aa459394baa2451ffacc02219ba54ef77ca44</Sha>
```

**Review by hoyosjs — APPROVED** [review]
Minor suggestion given the rollback

**Comment by ViktorHofer** [comment]
Reason for the package testing errors is https://github.com/dotnet/arcade/commit/582981cf4bc8de324d756b78c390c9610f20fb49#diff-4eed975465e70954faaaafdfff1b789ffb5346b1dbfde558f0d832e693eff73c which improves testing for RID specific assets. Had  to suppress false positive errors.

**Comment by ViktorHofer** [comment]
@Anipik this dependency update PR needs to go in before we branch off. We should hold branching off until CI is working again and the PR is merged.

**Comment by ViktorHofer** [comment]
CI is on the floor partially but is recovering now. The only change since the last run (which was mostly green) affects the AllConfigurations leg only which already passed. Merging to not require another build iteration.

**Pull request merged**